scripts/west_commands/runners/pyocd.py

```diff
@@ -125,6 +131,7 @@ def flash(self, **kwargs):
 
         cmd = ([self.pyocd] +
                ['flash'] +
+               self.erase_args +
```

Comment:
It looks like you can just put `-e auto` here and not add yet another argument. If the user wants to override, they can use `--flash-opts`, since that comes at the end.

Reply:
Right, but I dislike the fact that `-e sector` would then contend with `-e chip`, what guarantees that one or the other is actually enforced?

Reply:
Sorry, as you pointed out on slack, `-e sector`. My bad.

Reply:
Do we know for sure that using `--flash-opts` will override?

Reply:
Yes. I checked the pyocd code; it's a standard argparse-based parser argument which just updates the `erase` dest. I also built and installed a pyocd version with a printline thrown in for the final `_args.erase` value; it behaved as expected when specified multiple times (last one wins).

If that ever changes we can open a bug with pyocd or work around it in west, but having later instances of an option override the previous one is standard behavior unless otherwise specified in the help, so I'd rather not add yet another option given that we don't need to.

Reply:
Having later instances of an arg override previous is the expected behaviour for pyOCD that shouldn't change unless there is a future bug in argparse. The only arguments that don't behave that way are `--command` and `--pack`.

Reply:
Cool, thanks for verifying.